lib/s3_uploader.rb

```diff
@@ -0,0 +1,21 @@
+require 'aws-sdk'
+
+class S3Uploader
+  def self.upload(file, path)
+    s3_file = s3_bucket.objects[path].write(file: file)
+    s3_file.acl = :public_read
+
+    s3_file
+  end
+
+  def self.s3_bucket
+    bucket = s3.buckets[ENV['AWS_S3_BUCKET']]
+    bucket.acl = :public_read
+
+    bucket
+  end
+
+  def self.s3
+    @s3 ||= AWS::S3.new(access_key_id: ENV['AWS_ACCESS_KEY_ID'], secret_access_key: ENV['AWS_SECRET_ACCESS_KEY'])
```

Comment:
This could cause a race condition assigning an instance of aws s3 to a class, what if two people trying uploading to s3 simultaneously?

Always create an instance when using instance variables.

```rb
class S3Uploader
  attr_reader :file, :path, :options

  # Keep nice and simple public API
  def self.upload(file, path, options = {})
    # Make an instance internally and pass through the stuff
    uploader = new(file, path, options)
    uploader.upload
    uploader
  end

  def initialize(file, path, options = {})
    @file, @path, @options = file, path, options
  end

  def upload
    # ... do upload stuff in an instace
  end

  private
  
  def s3
    @s3 ||= # ...
  end
end
```